src/controller.cpp

```diff
@@ -602,7 +601,10 @@ void Controller::replace_feed(std::shared_ptr<RssFeed> oldfeed,
 	feed->set_tags(urlcfg->get_tags(oldfeed->rssurl()));
 	feed->set_order(oldfeed->get_order());
 	feedcontainer.feeds[pos] = feed;
-	enqueue_items(feed);
+	queueManager.autoenqueue(feed);
+	for (const auto& item : feed->items()) {
+		rsscache->update_rssitem_unread_and_enqueued(item, feed->rssurl());
```

Comment:
If I'm correct, this was taken out of `enqueue_items`, where we check three things before updating item: `!item->enqueued() && item->enclosure_url().length() > 0`, and then `utils::is_http_url(item->enclosure_url())` (Which could be written just as `!item->enqueued() && utils::is_http_url(item->enclosure_url()`). 

Is there a reason why this is omitted here?

Reply:
I don't do the checks because `update_rssitem_unread_and_enqueued` is harmless: if we run it on an item we didn't change in `autoenqueue`, it will simply write the values that are already in the DB.

I think I considered adding a comment here, but it would only be useful to reviewers who compare "before" and "after". For ordinary folks, this is just "dump feed item's "unread" and "enqueued" fields to the database".